client/src/components/pages/sign-up/index.tsx

```diff
@@ -13,10 +13,10 @@ import styles from './sign-up.module.scss';
 
 const SignUpGithubSchema = Yup.object().shape({
 	username: Yup.string()
-		.matches(/^[^\-][a-zA-Z0-9]{1,}\-?[a-zA-Z0-9]*[^\-]+$/, 'Use only letters, numbers and hyphens')
-		.required('Required')
 		.min(3, 'Minimum length: 3')
-		.max(20, 'Maximum length: 20'),
+		.max(20, 'Maximum length: 20')
+		.matches(/^[A-Za-z0-9]+\-?[A-Za-z0-9]+$/, 'Use only letters, numbers and hyphens')
```

Comment:
change this to 
[a-zA-Z0-9]+\-?[a-zA-Z0-9]+

According to https://regex101.com/, your validation string-size is always 3 symbols length. We need more